piet-gpu-hal/src/vulkan.rs

```diff
@@ -1421,26 +1450,35 @@ impl Layers {
     }
 }
 
-unsafe fn choose_compute_device(
+unsafe fn choose_device(
     instance: &Instance,
     devices: &[vk::PhysicalDevice],
-    surface: Option<&VkSurface>,
+    support_graphics: bool,
```

Comment:
It's slightly confusing to follow this because `support_present` is used as the variable name in the caller, which is not quite consistent. I recommend picking one name then optionally adding comments to discuss.

Reply:
Yes, since we need graphics support regardless, I think this parameter should just be removed. More comments on this on the next item.